org.eclipse.xtext.util/src/org/eclipse/xtext/util/ReplaceRegion.java

```diff
@@ -56,6 +58,25 @@ public void applyTo(StringBuilder builder) {
 		builder.replace(offset, getEndOffset(), text);
 	}
 	
+	@Override
+	public boolean equals(Object object) {
+		if (object instanceof ReplaceRegion) {
+			ReplaceRegion rr = (ReplaceRegion) object;
+
+			boolean equals = true;
+			equals = equals && Objects.equals(getOffset(), rr.getOffset());
+			equals = equals && Objects.equals(getLength(), rr.getLength());
+			equals = equals && Objects.equals(getText(), rr.getText());
+			return equals;
+		}
+		return false;
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(getOffset(), getLength(), getText());
```

Comment:
This builds an array of values including boxing conversion. I think the defaults as generated by Eclipse are better.